packages/transformers/react-refresh-babel/src/ReactRefreshBabelTransformer.js

```diff
@@ -0,0 +1,117 @@
+// @flow
+
+import semver from 'semver';
+import {Transformer} from '@parcel/plugin';
+import {relativeUrl} from '@parcel/utils';
+import SourceMap from '@parcel/source-map';
+import {transformFromAst} from '@babel/core';
+import generate from '@babel/generator';
+import {parse} from '@babel/parser';
+
+function shouldExclude(asset, options) {
+  return (
+    !asset.env.isBrowser() ||
+    !options.hot ||
+    !asset.isSource ||
+    asset.filePath.includes('packages/runtimes')
+  );
+}
+
+export default new Transformer({
+  canReuseAST({ast}) {
+    return ast.type === 'babel' && semver.satisfies(ast.version, '^7.0.0');
+  },
+
+  async parse({asset, options}) {
+    let code = await asset.getCode();
+    if (shouldExclude(asset, options)) {
+      return null;
+    }
+
+    return {
+      type: 'babel',
+      version: '7.0.0',
+      isDirty: false,
+      program: parse(code, {
+        filename: this.name,
+        allowReturnOutsideFunction: true,
+        strictMode: false,
+        sourceType: 'module',
+        plugins: [
+          'exportDefaultFrom',
+          'exportNamespaceFrom',
+          'dynamicImport',
+          'jsx'
+        ]
+      })
+    };
+  },
+
+  async transform({asset, options}) {
+    asset.type = 'js';
+    let ast = asset.ast;
+    if (!ast || shouldExclude(asset, options)) {
+      return [asset];
+    }
+
+    let reactRefreshBabelPlugin = (await options.packageManager.resolve(
+      'react-refresh/babel',
+      __filename
+    )).resolved;
+
+    let code = await asset.getCode();
+    let transformResult = transformFromAst(ast.program, code, {
+      code: false,
+      ast: true,
+      filename: asset.filePath,
+      babelrc: false,
+      configFile: false,
+      plugins: [reactRefreshBabelPlugin]
```

Comment:
Should we just add this plugin to the babel transformer like we have for JSX, TS, Flow, etc? Would we one less transform pass and a bunch less duplicated code.